### ISSUE TITLE
Optional setting to disable menu?

### ISSUE THREAD

**Issue by orschiro** [Ubuntu/gnome-shell-extension-appindicator#120]
Hey guys, 

Really like the implementation so far!

But the reason I still prefer and use [TopIcons](https://extensions.gnome.org/extension/495/topicons/) is the fact that I can click on an icon and it directly opens the window. With AppIndicator, however, I click the icon, it shows a menu with "Show/Hide window" from where I need to click again to actually open the window. 

Can we add a setting to disable this menu?

I prefer to save time and directly open the window. 

Or maybe directly show the window on right mouse click?

Thanks for listening!

**Comment by jhasse** [comment]
Depending on the app, middle or double-click should work. Have you tried that?

**Comment by orschiro** [comment]
@jhasse tried with Rambox and Slack. Both don't trigger any action when pressing middle or right mouse click.

**Comment by jhasse** [comment]
Those apps need to set a secondary action to show/hide the window. Without that it wouldn't help if we add an option to trigger this action on left click.

**Comment by orschiro** [comment]
What do you mean by secondary action?

Like Slack does with two actions they have?

One opening the main window, the other to open the preferences.

![image](https://user-images.githubusercontent.com/170947/36946349-e6c78098-1fbb-11e8-83b2-bf4911d919c9.png)

**Comment by jhasse** [comment]
No, I mean the secondary action apps can set via the API: https://bazaar.launchpad.net/~indicator-applet-developers/libappindicator/trunk.16.10/view/head:/src/app-indicator.h#L284

**Comment by hexauniverse** [comment]
Here's my solution:

After clicking on menu item dbus message are spawned, so we have an ability to send it manually. I've written a code for discord and mailspring, it's also makes left click behavior as middle-click. Paste it into [`_boxClicked` function](https://github.com/ubuntu/gnome-shell-extension-appindicator/blob/b50bdcbc54762326bd5c1fb646863a76a161921f/indicatorStatusIcon.js#L122):
```
        if (event.get_button() === 1) {
            if (this._indicator._proxy.Id === 'Genius') return;
            Main.panel.menuManager._closeMenu(true, Main.panel.menuManager.activeMenu);
            if (this._indicator._proxy.Id === 'Mailspring1') {
                imports.gi.GLib.spawn_command_line_sync(`
                    dbus-send
                        --type=method_call --dest=${this._menuClient._busName}
                        /org/ayatana/NotificationItem/Mailspring1/Menu
                        com.canonical.dbusmenu.Event
                        int32:2 string:"clicked" variant:int32:0 uint32:0
                `);
                return;
            }
            if (this._indicator._proxy.Id === 'discord1') {
                imports.gi.GLib.spawn_command_line_sync(`
                    dbus-send
                        --type=method_call --dest=${this._menuClient._busName}
                        /org/ayatana/NotificationItem/discord1/Menu
                        com.canonical.dbusmenu.Event
                        int32:4 string:"clicked" variant:int32:0 uint32:0
                `);
                return;
            }
            this._indicator.open();
            return;
        }
```

You can use `dbus-monitor` to find dbus messages for your programs
